### ISSUE TITLE
MXML Components Explorer

### ISSUE THREAD

**Issue by joshtynjala** [BowlerHatLLC/feathers-sdk#15]

**Comment by teotigraphix** [comment]
Would you want help with this?

**Comment by joshtynjala** [comment]
Sure! That would save me some time.

**Comment by teotigraphix** [comment]
Maybe if you got it started so I can see how you want it structured, then I could start working on each Screen and do a pull request?

**Comment by joshtynjala** [comment]
Yeah, that seems like a good idea. I'll try to get it started in the next week.

**Comment by teotigraphix** [comment]
I'm using it pretty extensively in my apps now(ported most view AS to MXML), so giving it a good test. :)

**Comment by joshtynjala** [comment]
Great! Sounds like things are pretty stable then.

**Comment by teotigraphix** [comment]
Knock on wood I have only had one problem and that was creating a subclass of LayoutGroup IE VGroup, then trying to use the VGroup as a base class in MXML.

The compiler barfed and said the my MXML needed to be a subclass of starling.events.EventDispatcher, which is was.

I need to file that bug in this repo actually.

**Comment by joshtynjala** [comment]
That sounds like a bug that I fixed in this commit: bd8049c932a018e1a569483bc4bc5aac99fdb9ad

Did you update to the stable build of 2.2.0?

**Comment by teotigraphix** [comment]
AH! Actually, I just did today ironically. So I will try it again and let you know.

**Comment by teotigraphix** [comment]
Yes, seems to be fixed, right on!

**Comment by quick6black** [comment]
HI Josh, I think this is related to a component explorer. Would you be able to provide an example of how to use the drawer component in MXML? thanks

**Comment by joshtynjala** [comment]
You can use `<f:DrawersApplication>` if you want the Drawers at the root of your display list:

```
<?xml version="1.0" encoding="utf-8"?>
<f:DrawersApplication xmlns:f="library://ns.feathersui.com/mxml"
    xmlns:fx="http://ns.adobe.com/mxml/2009"
    theme="feathers.themes.MetalWorksMobileTheme">

    <f:leftDrawer>
        <f:PanelScreen title="Drawer">
        </f:PanelScreen>
    </f:leftDrawer>

    <f:PanelScreen title="Content">
    </f:PanelScreen>

</f:DrawersApplication>
```

The topDrawer, rightDrawer, bottomDrawer, and leftDrawer properties let you set the drawers on each side. I made the content property the default property for MXML, so you you don't need to actually type out `<f:content></f:content>`. Just specify any single component as a child element, and it will be the content.

`<f:Drawers>` works pretty much the same way, if you don't want it to be the root of your application.

**Comment by quick6black** [comment]
awesome, thanks for the help.
